contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -2103,12 +2103,13 @@ bigint_power(PG_FUNCTION_ARGS)
 	Numeric		arg2 = PG_GETARG_NUMERIC(1);
 	int64 result;
 	Numeric		arg1_numeric,
+				result_floor,
 				result_numeric;
 
 	arg1_numeric = DatumGetNumeric(DirectFunctionCall1(int8_numeric, arg1));
 	result_numeric = DatumGetNumeric(DirectFunctionCall2(numeric_power, NumericGetDatum(arg1_numeric), NumericGetDatum(arg2)));
-
-	result = DatumGetInt64(DirectFunctionCall1(numeric_int8, NumericGetDatum(result_numeric)));
+	result_floor = DatumGetNumeric(DirectFunctionCall1(numeric_floor, NumericGetDatum(result_numeric)));
```

Comment:
Applicable for other functions also

Reply:
Converting Numeric to int8 is rounding of the results to the nearest integer. Example - power(2,-1) result is 0.5000, if we directly convert it into int8 it will be rounded of to 1, but the correct answer is 0. That's why, used the numeric_floor function before the conversion.